Automattic/wp-calypso
# BBE Content Form: add search terms input

Fixes Automattic/martech#1618

## Proposed Changes

* This change adds the search terms input introduced in https://github.com/Automattic/wp-calypso/pull/73705 to the content form.
* This is a required field and cannot be skipped.
* This PR also renames (both copy and internal) the "Site Logo" section to the "Site Information" section.
<img width="809" alt="image" src="https://user-images.githubusercontent.com/5436027/224031077-ae24dd92-8751-4df1-8b37-8a92d9579ab0.png">



## Testing Instructions


* Apply D104171-code
* Go to `/start/do-it-for-me`.
* On the site options step, enter some text in the search terms text box. 
* Proceed through the flow steps and complete the purchase.
* Once you reach the website content form, confirm that you can see the new section at the top.
* Confirm that the search text you entered before checkout is present.
* Confirm that the save and submit actions work correctly.

## Pre-merge Checklist
